### DISCUSSION TITLE
Multiple projects in one repo

### DISCUSSION THREAD

**Discussion by dan-haynes** [CFC-Servers/GLuaTest#53] [Q&A] [answered]
Feel free to correct me if I'm totally wrong here but it seems like the testing with github actions/docker is really meant for single addon projects. For our use case, we have a github that has a lot of our production code on it in the format of repo/garrysmod/addons and repo/garrysmod/gamemode. I tried to repurpose a lot of the code to work in our format but it seems a lot of the existing code relies on the structure of it just containing one addon. Any sense of direction would be great as this is a very helpful tool in our development and we have used it a ton to make our code more maintainable.

**Comment by sarahsturgeon** [comment]
Hi @zeeptin!

You bring up a good point, I definitely didn't create GLuaTest with this usecase in mind.

Honestly I didn't realize that so many people did it this way. I'd love to make GluaTest work for this configuration too.

So I'm thinking of a short term solution that should help:
- Add more flexibility in how projects can be structured for the docker/github action runs

I think as long as we can get the gamemode/addons directories in the right place, you should be able to add your gamemode-specific tests somewhere that makes sense to you, and then each addon can have their tests in the normal place.

Does that sound like it would make life easier for your setup?

**Reply by dan-haynes** [reply]
Yeah, that should work. Sorry, tried to do as much as I could on a fork but was misunderstanding some of the project folder structure, I think. I was thinking maybe if we could change the volume mount for the gmod server files, it would work but I think it is a little more complex with the "project" files and requirements, etc. I would be happy to hop on a discord call if anything needs to be clarified. (Zeeptin is my discord).

**Reply by sarahsturgeon** [reply]
Great!

Yeah I can understand being confused. The codebase is a little sloppy in places.

 I don't think it'll be a 4-line change or anything, but I'm certain we can get it done.

I'll reach out to you on Discord - at minimum it would be very helpful if you could help me build a local example project that's structured like yours.

For the sake of being thorough, I'll also report back to this Discussion when we have something solidified.

**Comment by sarahsturgeon** [comment, marked as answer]
With the release of https://github.com/CFC-Servers/GLuaTest/releases/tag/0.27, this is now addressed 👍 

Thanks again for your help testing and refining!
